server/src/main/java/org/apache/cloudstack/direct/download/DirectDownloadManagerImpl.java

```diff
@@ -362,7 +366,13 @@ private Answer sendDirectDownloadCommand(DirectDownloadCommand cmd, VMTemplateVO
         }
         if (!downloaded) {
             logUsageEvent(template, poolId);
-            throw new CloudRuntimeException("Template " + template.getId() + " could not be downloaded on pool " + poolId + ", failing after trying on several hosts");
+            if (!answerDetails.isEmpty()){
+                Account caller = CallContext.current().getCallingAccount();
+                if (caller != null && caller.getType() == Account.ACCOUNT_TYPE_ADMIN){
```

Comment:
should this be
```suggestion
                if (caller != null && caller.getType() == Account.Type.ADMIN){
```
?

Reply:
I think this was refactored b/w 4.16-4.18.